components/org.wso2.carbon.sp.jobmanager.core/src/main/java/org/wso2/carbon/sp/jobmanager/core/appCreator/SafeZkClient.java

```diff
@@ -28,13 +33,45 @@
  */
 class SafeZkClient {
 
+    private ZkClient zkClient;
+    private ZkUtils zkUtils;
+    private static final Logger log = Logger.getLogger(SafeZkClient.class);
+
     /**
      * Creates a new Zookeeper client for the specified server URL.
      *
-     * @param zooKeeperServerUrl server URL
+     * @param zooKeeperServerUrls server URL
      * @return a new Zookeeper client
      */
-    public ZkClient createZkClient(String zooKeeperServerUrl) {
-        return new ZkClient(zooKeeperServerUrl, 10000, 8000, ZKStringSerializer$.MODULE$);
+    public ZkUtils createZkClient(String[] zooKeeperServerUrls, boolean isSecureKafkaCluster) {
+        for (String zooKeeperServerUrl : zooKeeperServerUrls) {
+            try {
+                zkClient = new ZkClient(zooKeeperServerUrl, 10000, 5000, ZKStringSerializer$.MODULE$);
```

Comment:
Why are we hard-coding these?

Reply:
Made user configurable in commit b2d4301